tests/image-stream/mime.spec.js

```diff
@@ -0,0 +1,100 @@
+
+  describe('.getMimeTypeFromFileName()', function() {
+
+    it('should resolve application/x-bzip2 for a bz2 archive', function() {
+      const file = path.join(DATA_PATH, 'bz2', 'etcher-test.img.bz2');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/x-bzip2');
+      });
+    });
+
+    it('should resolve application/x-xz for a xz archive', function() {
+      const file = path.join(DATA_PATH, 'xz', 'etcher-test.img.xz');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/x-xz');
+      });
+    });
+
+    it('should resolve application/gzip for a gz archive', function() {
+      const file = path.join(DATA_PATH, 'gz', 'etcher-test.img.gz');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/gzip');
+      });
+    });
+
+    it('should resolve application/zip for a zip archive', function() {
+      const file = path.join(DATA_PATH, 'zip', 'zip-directory-etcher-only.zip');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/zip');
+      });
+    });
+
+    it('should resolve application/octet-stream for an uncompressed image', function() {
+      const file = path.join(DATA_PATH, 'images', 'etcher-test.img');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/octet-stream');
+      });
+    });
+
+    it('should resolve application/x-iso9660-image for an uncompressed iso', function() {
+      const file = path.join(DATA_PATH, 'images', 'etcher-test.iso');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/x-iso9660-image');
+      });
+    });
+
+    it('should resolve application/x-apple-diskimage for a compressed Apple disk image', function() {
+      const file = path.join(DATA_PATH, 'dmg', 'etcher-test-zlib.dmg');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/x-apple-diskimage');
+      });
+    });
+
+    it('should resolve application/x-apple-diskimage for an uncompressed Apple disk image', function() {
+      const file = path.join(DATA_PATH, 'dmg', 'etcher-test-raw.dmg');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/x-apple-diskimage');
+      });
+    });
+
+    it('should resolve application/octet-stream for an unrecognized file type', function() {
+      const file = path.join(DATA_PATH, 'unrecognized', 'random.rpi-sdcard');
+      return mime.getMimeTypeFromFileName(file).then((type) => {
+        m.chai.expect(type).to.equal('application/octet-stream');
+      });
+    });
+
+    it('should resolve the actual MIME type, without caring about the extension', function() {
```

Comment:
nit: it _does_ care about the extension, because it only checks the first 262 bytes if it doesn't recognise the extension. Perhaps something like `should resolve the actual MIME type for an unrecognized extension` ?
Is there any danger that a future version of the `mime-types` module might recognise `*.md` as `text/markdown` ?